scripts/run-live-build.sh

```diff
@@ -129,36 +129,64 @@ while ! curl --output /dev/null --silent --head --fail \
 done
 set -o errexit
 
+pkg_mirror_secondary=''
 if [[ -n "$DELPHIX_PACKAGE_MIRROR_SECONDARY" ]]; then
-	sed "s|@@URL@@|$DELPHIX_PACKAGE_MIRROR_SECONDARY|" \
-		<config/archives/delphix-secondary-mirror.list.in \
-		>config/archives/delphix-secondary-mirror.list
+	pkg_mirror_secondary="$DELPHIX_PACKAGE_MIRROR_SECONDARY"
+else
+	#
+	# If no secondary package mirror is provided, then pull in the latest
+	# mirror dataset for the build. If no latest dataset is found, then fail.
+	#
+	source_url="http://linux-package-mirror.delphix.com/$UPSTREAM_PRODUCT_BRANCH/latest/"
+	pkg_mirror_secondary="$(curl -Ls -o /dev/null -w '%{url_effective}' $source_url)ppas"
```

Comment:
Can you explain the design for this, and/or point me to any existing written documentation for how this is supposed to work? What is the "public interface" that we're consuming here? I'm not questioning the validity, I would just like to better understand the full solution; e.g. how is this "service" managed, how will this be updated, what is the "interface" that we're adhering too, etc.

Is there a design doc that I can look over?

Reply:
We have nginx running a web server on our internal package mirror. On every successful update to the mirror we update an nginx config file to redirect  `http://linux-package-mirror.delphix.com/<delphix-branch>/latest/` to the latest updated mirror dataset, which can be dereferenced with a curl command as seen here. There is a redirect rule per branch. This mirror update and rewrite change are both done through this jenkins job `http://ops.jenkins.delphix.com/job/devops-gate/job/master/job/linux-package-mirror-sync/job/master/job/mirror1-sync/` (which is currently disabled to freeze the updates). The rewrite rule update in the job is here `https://gitlab.delphix.com/devops/devops-gate/blob/master/jenkins/jobs/pipelines/linux_package_mirror_sync.groovy#L402`. I am not sure what you mean by "public interface", let me know if that answers your questions, or if I can clarify more.

We do not have a design doc yet, but there should be one soon.

Reply:
Thanks for the details.

After the "latest" redirect has changed, how long will the previous URL work? I presume, if it changes in the middle of a running build, the old build URL(s) will still work, right? For how long will they work? and what will be the cadence for updating the redirect?

Reply:
Since we are dereferencing the redirect and we are not removing the previous dataset, the previous URL will continue to work indefinitely at the moment. We will eventually add clean up for old datasets, the plan being to remove old datasets after a few days so that we do not break any in flight builds.